### ISSUE TITLE
COMError with GetAllDevices (CreateDevice)

### ISSUE THREAD

**Issue by hovissimo** [AndreMiras/pycaw#15]
I'm playing with pycaw with the eventual goal of being able to reassign the default audio device (I'm not sure that's even supported yet). All the same, I can't enumerate the devices using GetAllDevices because I get a COMError. I get the same error when running the unit tests.  I have no experience using Windows APIs or programming in C++, but if I can do anything to help troubleshoot this I'll be happy to help.

- current requirements (pip freeze)
```
comtypes==1.1.7
enum34==1.1.6
future==0.17.1
psutil==5.6.1
pycaw==20181226
```
- OS version
`10.0.17763 Build 17763`
- architecture
`x64`
- Python version
`3.7.2`
- pycaw version
`v20181226`
- project files
```python
# this is sufficient to reproduce the error, or run the tests
from pycaw.pycaw import AudioUtilities

devices = AudioUtilities.GetAllDevices()
```
- exact stacktrace / accurate error message
```
C:\Users\hovis\dev\pycaw>python -m unittest discover
E..F..
======================================================================
ERROR: test_device_unicode (tests.test_core.TestCore)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\hovis\dev\pycaw\tests\test_core.py", line 44, in test_device_unicode
    devices = AudioUtilities.GetAllDevices()
  File "C:\Users\hovis\dev\pycaw\pycaw\pycaw.py", line 732, in GetAllDevices
    devices.append(AudioUtilities.CreateDevice(dev))
  File "C:\Users\hovis\dev\pycaw\pycaw\pycaw.py", line 704, in CreateDevice
    value = store.GetValue(pk)
_ctypes.COMError: (-536870389, None, (None, None, None, 0, None))

======================================================================
FAIL: test_audio_endpoint_volume_example (tests.test_examples.TestExamples)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\hovis\dev\pycaw\tests\test_examples.py", line 38, in test_audio_endpoint_volume_example
    lines[1], 'volume.GetMasterVolumeLevel(): -20.0')
AssertionError: 'volume.GetMasterVolumeLevel(): -26.715423583984375' != 'volume.GetMasterVolumeLevel(): -20.0'
- volume.GetMasterVolumeLevel(): -26.715423583984375
?                                  ^ ^^^^^^^^^^^^^^^
+ volume.GetMasterVolumeLevel(): -20.0
?                                  ^ ^


----------------------------------------------------------------------
Ran 6 tests in 0.495s

FAILED (failures=1, errors=1)

C:\Users\hovis\dev\pycaw>

```
- and so on

**Comment by MennovDijk** [comment]
Same issue here...

**Comment by micolous** [comment]
This looks to be similar to #3, and I'm having the same issue on Python 3.9 with Windows 10 v2004 (Build 19041.746).

I had a poke at what went wrong by adding some logging statements to `CreateDevice`:

```python
           for j in range(propCount):
                pk = store.GetAt(j)
                try:
                    value = store.GetValue(pk)
                    v = value.GetValue()
                except:
                    # error getting property
                    print(f'Error getting device {id} property {pk}')
                    print('other properties:')
                    print(properties)
                    continue
                # TODO
                # PropVariantClear(byref(value))
                name = str(pk)
                properties[name] = v
```

For me, it failed to get the property `{A45C254E-DF1C-4EFD-8020-67D146A850E0} 14`, which is interesting because that property appears to be `DEVPKEY_Device_FriendlyName` -- which seems like it shouldn't ever fail!

There were multiple devices with this problem for me, which showed up regardless of whether they were actually plugged in to the machine and/or explicitly disabled at the time of calling it:

* NVIDIA's HDMI or DisplayPort audio outputs (for various displays previously connected to the machine)
* Logitech C930e USB webcam (which has a UAC device for the microphone, using Microsoft's class driver)
* Two models of HDMI capture devices (which have a UAC device for the audio input, using Microsoft's class driver)

Interestingly there seems to be multiple entries for the _same_ device, even the disconnected ones, of which _some_ report their name correctly and some don't.  For example, the HDMI capture device (which uses Microsoft's UAC driver) has two entries, the one that reports correctly has the property:

`'{83DA6326-97A6-4088-9453-A1923F573B29} 3': 'wdma_usb.inf:ed86ca114a9cb1b5:USBAudio:10.0.19041.388:USB\\Class_01',`

But the one that doesn't report correctly has:

`'{83DA6326-97A6-4088-9453-A1923F573B29} 3': 'wdma_usb.inf:ed86ca114a9cb1b5:USBAudio:10.0.18362.997:USB\\Class_01',`

The broken one appears to correspond to an older build of the `USBAudio` driver, whereas the good one looks like it comes from the build of Windows 10 I'm running (19041.746).

It looks like _anyone_ could trigger this, even using Microsoft's drivers, and `pycaw` should handle a failure to get properties more gracefully.

**Comment by AndreMiras** [comment]
This got addressed in #30 which is merged to develop. Thanks @hovissimo for the initial work.
Closing
